WcaOnRails/app/models/competition.rb

```diff
@@ -658,20 +660,16 @@ def started?
     !start_date.nil? && start_date < Date.today
   end
 
-  def country_name
-    country ? country.name : nil
-  end
-
-  def country
-    Country.find_by_id(countryId)
-  end
-
   def organizers_or_delegates
     self.organizers.empty? ? self.delegates : self.organizers
   end
 
   def continent
-    country ? Continent.find_by_id(country.continentId) : nil
+    # NOTE: Doing this find is ok right now, because it's only called once when
+    # sending reports. However if at some point we want to include the continent
+    # in some competitions list, we should define an association the same way
+    # we do for country to avoid "hidden" N+1 query.
+    Continent.find_by_id(country.continentId)
```

Comment:
I think there's no reason for keeping this. We could just do `has_one :continent, through: :country`, right?